File: lib/urls.dart
class Urls {
  static const String baseUrl = "http://127.0.0.1:8000";

  static String createTask() => "$baseUrl/tasks/add-task/";
  static String getTasks() => "$baseUrl/tasks/";
  static String getSingleTask(String id) => "$baseUrl/tasks/$id/";
  static String updateTask(String id) => "$baseUrl/tasks/$id/update/";
  static String deleteTask(String id) => "$baseUrl/tasks/$id/delete/";
}

File: lib/backend/interview_todo_app/todo_backend/urls.py
from django.urls import include, path
from . import views

urlpatterns = [
    path('', views.getRoutes),

    path('tasks/add-task/', views.add_task),
    path('tasks/', views.get_tasks),
    path('tasks/<str:pk>/', views.get_single_task),
    path('tasks/<str:pk>/update/', views.update_task),
    path('tasks/<str:pk>/update-task-status/', views.update_task_status),
    path('tasks/<str:pk>/delete/', views.delete_task),
]
